### PR TITLE
Modifying a few things in the tutorial and adding the write/read part

### PR DESCRIPTION
**Pull request by MRegeard**
PR that add the writing and reading part of the tutorial by modifying the HDU table. 

@alvmas and @jalu98 can you try to run the tutorial because I was not able to do it myself. 

If you have any comments/suggestion ! 

Signed-off-by: Maxime Regeard <regeard@apc.in2p3.fr>

### CODE REVIEW

**Comment by MRegeard** [comment]
Hi there, 

I think I will add more comment to some part of the tutorial. So I will probably push another commit in a few days. 

By the way, I have talked with Bruno (from gammapy dev team) that told me that it would be better to choose one of the two methods for computing phase since our recipe is pretty long, or let the reviewer decide for us (I don't think we want that). 

Moreover I have talk with PINT people (see https://github.com/nanograv/PINT/issues/1504 and https://github.com/nanograv/PINT/pull/1511) so they have created a function that does exactly what we want to do and its really straight forward !! And from what @jalu98 has added in the last PR I think we should go for the 'PINT internal function' one. 

Tell me what you think about that. 

Maybe we should meet at some point to finish this recipe.

**Comment by alvmas** [comment]
Hi,
So that new function basically that all the corrections that we need for the analysis using Time objects as input. So, yes, it is basically what you did but with this function is should be more compact.

So if we need to decide for one method, why not using your method but instead of applying all the corrections manually, directly using that new function? Or is it not implemented yet?

**Comment by MRegeard** [comment]
So in the last commit I have modified this part of the recipe to use `get_TOAs_list()`function instead of calling them one by one. This was suggested by PINT people in the PR that I put above. The function that there are "designing for us" and which create a `TOAs` object directly from a astropy.Time object is still in dev so I think we should stick with this until it is release.

**Comment by alvmas** [comment]
HI @MRegeard,
So I will run the notebook to see it works well. On a first fast check I can see that the directory of the new hdu table is not the same as the pulsar files. Wouldn't be easier if we keep the hdu file in the same directory as the rest of the files?

Second, finally do you agree to maintain the second way in the recipe and delete the ".tim method" as gammapy developers suggested to keep only onw?

**Comment by MRegeard** [comment]
Hi @alvmas, 
I think that you can put the hud table in the directory of the new pulsar files. But then you must go to all the IRFs and modify their "FILE_DIR" entry in the new hdu table to add "../" instead of "./" . I think here the logic is to have different hdu table in the main data store directory and then to point to different directory / files in these hdu table. Moreover, doing this we don't have to copy the obs-index.fits table in the new directory. 

Yes I think now that we are sure that we are doing things properly with PINT (as I checked with them) and since they are going to put this function, we should go with the second method and drop the .tim file method.

**Comment by alvmas** [comment]
I checked the notebook. I tried to use pint version 0.9.3, which is going to be the most recent, and the function toa.TOA is raising an error when trying to compute the shapiro delay.

File: recipes/pulsar_analysis/Pulsar_analysis_gammapy.ipynb [hunk 1]
```diff
@@ -588,302 +617,617 @@
...
+    "                for t in times)\n",
     "\n",
-    "ts.compute_TDBs(ephem='DE421')\n",
-    "ts.compute_posvels(ephem='DE421',planets=True)\n",
-    "        \n",
-    "phss = model.phase(ts,abs_phase=True)[1]\n",
-    "    \n",
-    "phases_2 = np.where(phss < 0.0 , phss + 1.0 , phss)"
+    "# create a TOAs object from a list of TOA\n",
+    "ts = toa.get_TOAs_list(toa_list, ephem='DE421', include_gps=include_gps, include_bipm=include_bipm) \n",
```
Comment (alvmas) [inline recipes/pulsar_analysis/Pulsar_analysis_gammapy.ipynb]: include planets=True

File: recipes/pulsar_analysis/Pulsar_analysis_gammapy.ipynb [hunk 2]
```diff
@@ -588,302 +617,617 @@
...
-    "include_gps = True\n",
     "\n",
-    "toa_list = list(toa.TOA(t, 4.5, obs='magic', ephem='DE421', include_bipm=str(include_bipm), \n",
-    "                                include_gps=str(include_gps)) for t in times)\n",
-    "        \n",
-    "ts = toa.TOAs(toalist=toa_list)\n",
-    "ts.clock_corr_info={'include_bipm': str(include_bipm), 'include_gps': str(include_gps)}\n",
+    "# put this to True is your observatory has clock correction files. \n",
+    "# if it is set to True but your observatory does not have clock correction files, it will be ignored.\n",
+    "include_bipm = False\n",
```
Comment (alvmas) [inline recipes/pulsar_analysis/Pulsar_analysis_gammapy.ipynb:541]: Chane False by 'False'. Same for include_gps

File: recipes/pulsar_analysis/Pulsar_analysis_gammapy.ipynb [hunk 3]
```diff
@@ -903,22 +1247,24 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 493,
+   "execution_count": null,
    "id": "a3251096",
    "metadata": {},
    "outputs": [],
    "source": [
+    "# define on and off region within the phase space\n",
```
Comment (alvmas) [inline recipes/pulsar_analysis/Pulsar_analysis_gammapy.ipynb]: This last part can be deleted (to be include in other notebook)

**Review by alvmas — COMMENTED** [review]
These are the errors that I found when trying to execute the notebook

**Comment by MRegeard** [comment]
Done !
I think the on the code side we are pretty good with this.
But we might need to add more "text" explanations.

**Review by alvmas — APPROVED** [review]

**Pull request merged**